lib/Widen.php

```diff
@@ -167,4 +167,52 @@ public function list_collections( $type = 'global' ) {
 			return $data;
 		}
 	}
+
+	/**
+	 * Returns a specific image size using widen's templated url.
+	 *
+	 * @param string $templated_url The templated url provided by Widen. This is returned in the image response.
+	 * @param int    $width         The target image width.
+	 * @param int    $height        The target image height.
+	 * @param int    $scale         The scale of the image.
+	 */
+	public static function create_url_from_template( $templated_url, $width = null, $height = null, $scale = 1 ): string {
+		$url = Util::sanitize_image_url( $templated_url );
+
+		if ( $width && $height ) {
+			$size = $width . 'x' . $height;
+		} elseif ( $width ) {
+			$size = $width;
+		} else {
+			$size = 'exact';
+		}
+
+		// Perform template string replacements.
+		$url = str_replace( '{size}', $size, $url );
+		$url = str_replace( '{scale}', $scale, $url );
```

Comment:
This is where we are manipulating the templated URL I discovered.

Widen's API returns an image size named "templated" and it includes the following URL structure:
```
https://embed.widencdn.net/img/masonite/8tucbhveew/{size}px@{scale}x/VSG-926-built_SW6720.jpg
```

And to my surprise, when using 'exact' in this template, it does infact return the full size image even though it looks goofy with a px.

i.e. https://embed.widencdn.net/img/masonite/8tucbhveew/exactpx@1x/VSG-926-built_SW6720.jpg